services/src/main/java/org/keycloak/services/resources/admin/permissions/MgmtPermissions.java

```diff
@@ -98,17 +112,28 @@ class MgmtPermissions implements AdminPermissionEvaluator, AdminPermissionManage
 
     private void initIdentity(KeycloakSession session, AdminAuth auth) {
         final String issuedFor = auth.getToken().getIssuedFor();
-
-        if (Constants.ADMIN_CLI_CLIENT_ID.equals(issuedFor) || Constants.ADMIN_CONSOLE_CLIENT_ID.equals(issuedFor)) {
-            this.identity = new UserModelIdentity(auth.getRealm(), auth.getUser());
-        } else {
-            ClientModel client  = session.clients().getClientByClientId(auth.getRealm(), issuedFor);
-            if (client != null && Boolean.parseBoolean(client.getAttribute(Constants.SECURITY_ADMIN_CONSOLE_ATTR))) {
-                this.identity = new UserModelIdentity(auth.getRealm(), auth.getUser());
-            } else {
-                this.identity = new KeycloakIdentity(auth.getToken(), session);
+        AccessToken accessToken = auth.getToken();
+        //support for lightweight access token
+        if (auth.getToken().getSubject() == null) {
+            //get user session
+            UserSessionProvider sessions = session.sessions();
+            UserSessionModel userSession = sessions.getUserSession(adminsRealm, auth.getToken().getSessionId());
+            if (userSession == null) {
+                userSession = sessions.getOfflineUserSession(adminsRealm, auth.getToken().getSessionId());
             }
+
+            //get client session
+            ClientModel client = adminsRealm.getClientByClientId(issuedFor);
```

Comment:
Same comment like the other place for `userSession`, `client` and `clientSession` null checks.